massiv/src/Data/Array/Massiv/Stencil.hs

```diff
@@ -88,14 +90,15 @@ makeStencil
   => Border e -- ^ Border resolution technique
   -> ix -- ^ Size of the stencil
   -> ix -- ^ Center of the stencil
-  -> ((ix -> Elt e) -> Elt a) -- ^ Stencil function that receives another function as
+  -> (forall f. Applicative f => (ix -> f e) -> f a)
```

Comment:
This is not gonna work.There is no way use anything but `Functor` and `Applicative`. For example I cannot use `Num` when making a stencil, so this will not compile:

```haskell
avgStencil :: Stencil Ix1 Float Float
avgStencil = makeStencil Edge 3 1 $ \ get ->
  (  get (-1) + get 0 + get 1 ) / 3
```

The fact that `Identity`, which is an instance of `Num`, is used later, doesn't help us at all. A user would be forced to write it in `Applicative` style:

```haskell
avgStencil :: Stencil Ix1 Float Float
avgStencil = makeStencil Edge 3 1 $ \ get ->
  liftA2 (/) (liftA2 (+) (liftA2 (+) (get (-1)) (get 0)) (get 1)) (pure 3)
```